Assets/MRTK/SDK/Experimental/Drawstring/Scripts/TwoHandleManipulator.cs

```diff
@@ -0,0 +1,937 @@
+                rightHandle.localPosition = Vector3.right * elasticDistance / 2.0f;
+
+                UpdateLineData();
+            }
+        }
+
+        #endregion MonoBehaviour Functions
+
+        #region Private Methods
+        private Vector3 GetPointersGrabPoint()
+        {
+            Vector3 sum = Vector3.zero;
+            int count = 0;
+            foreach (var p in pointerIdToPointerMap.Values)
+            {
+                sum += p.GrabPoint;
+                count++;
+            }
+            return sum / Math.Max(1, count);
+        }
+
+        private MixedRealityPose GetPointersPose()
+        {
+            Vector3 sumPos = Vector3.zero;
+            Vector3 sumDir = Vector3.zero;
+            int count = 0;
+            foreach (var p in pointerIdToPointerMap.Values)
+            {
+                sumPos += p.pointer.Position;
+                sumDir += p.pointer.Rotation * Vector3.forward;
+                count++;
+            }
+
+            return new MixedRealityPose
+            {
+                Position = sumPos / Math.Max(1, count),
+                Rotation = Quaternion.LookRotation(sumDir / Math.Max(1, count))
+            };
+        }
+
+        private Vector3 GetPointersVelocity()
+        {
+            Vector3 sum = Vector3.zero;
+            int numControllers = 0;
+            foreach (var p in pointerIdToPointerMap.Values)
+            {
+                // Check pointer has a valid controller (e.g. gaze pointer doesn't)
+                if (p.pointer.Controller != null)
+                {
+                    numControllers++;
+                    sum += p.pointer.Controller.Velocity;
+                }
+            }
+            return sum / Math.Max(1, numControllers);
+        }
+
+        private Vector3 GetPointersAngularVelocity()
+        {
+            Vector3 sum = Vector3.zero;
+            int numControllers = 0;
+            foreach (var p in pointerIdToPointerMap.Values)
+            {
+                // Check pointer has a valid controller (e.g. gaze pointer doesn't)
+                if (p.pointer.Controller != null)
+                {
+                    numControllers++;
+                    sum += p.pointer.Controller.AngularVelocity;
+                }
+            }
+            return sum / Math.Max(1, numControllers);
+        }
+
+        private bool IsNearManipulation()
+        {
+            foreach (var item in pointerIdToPointerMap)
+            {
+                if (item.Value.IsNearPointer)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+        #endregion Private Methods
+
+        #region Public Methods
+
+        /// <summary>
+        /// Releases the object that is currently manipulated
+        /// </summary>
+        public void ForceEndManipulation()
+        {
+            // end manipulation
```

Comment:
Nit: This comment doesn't add that much - either delete it or expand upon it.

https://testing.googleblog.com/2017/07/code-health-to-comment-or-not-to-comment.html